Classes/common/CDTFetchChanges.m

```diff
@@ -79,12 +90,46 @@ - (void)main
                                                            options:&options
                                                             filter:nil
                                                             params:nil];
-        lastSequence = [self notifyChanges:changes startingSequence:lastSequence];
-    } while (changes.count > 0);
+        if (!thereIsALimit) {
+            // Limit not set. Get as many results as we can.
+            lastSequence = [self notifyChanges:changes startingSequence:lastSequence];
+
+            doLoop = (changes.count > 0);
+        } else if (mResultsLimit > 0) {
+            // There is a limit but we have not reached it yet.
+            lastSequence = [self notifyChanges:changes startingSequence:lastSequence];
+
+            if (changes.count == 0) {
+                // We did not reach the limit but there are not more results. Thefore there are
+                // not more data coming and we can stop looping.
+                doLoop = NO;
+            } else {
+                // Subtract the results so far
+                mResultsLimit -= [changes count];
+
+                if (mResultsLimit == 0) {
+                    // We reached the limit but we need to know if there are more data coming.
+                    // Loop again
+                    options.limit = 1;
```

Comment:
I mean, I have to return at most the number of results specified in `resultsLimit` and the `newSequenceValue` has to be equal to the sequence of the last result returned. As I see it I have no other options but to make one last `select` to see if there are more data after `newSequenceValue`.